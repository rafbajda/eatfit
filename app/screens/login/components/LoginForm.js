import React from 'react';
import { Item, Input, Content, Text } from 'native-base';
import { Formik } from 'formik';
import { DotIndicator, UIActivityIndicator } from 'react-native-indicators';
import {
    CenterRow,
    CenterFormContainer,
    ErrorText
} from '../../../shared/styles/common';
import styles, {
    ForgotPasswordText,
    LoginButton
} from '../styles/loginFormStyles';
import { globalGreen, globalWhite } from '../../../shared/constants/colors';
import screens from '../../../navigation/screens';
import validationSchemas from '../../../shared/utils/validationSchemas';
import formInitialValues from '../../../shared/constants/formInitialValues';
import I18n from 'i18n-js';

const LoginForm = props => {
    const {
        login,
        isKeyboardVisible,
        authLoading,
        nav,
        additionalTopPadding,
        loginInitialValues,
        loginSchema
    } = {
        ...props,
        ...styles,
        ...formInitialValues,
        ...validationSchemas
    };

    const { t } = I18n;

    return (
        <Formik
            initialValues={loginInitialValues}
            onSubmit={values => login(values)}
            validationSchema={loginSchema}
        >
            {formikProps => (
                <Content>
                    {authLoading ? (
                        <DotIndicator color={globalGreen} count={5} />
                    ) : null}
                    <CenterRow>
                        <CenterFormContainer
                            style={
                                isKeyboardVisible ? additionalTopPadding : null
                            }
                        >
                            <Item regular>
                                <Input
                                    keyboardType="email-address"
                                    placeholder={t('placeholders.email')}
                                    onChangeText={formikProps.handleChange(
                                        'email'
                                    )}
                                    onBlur={formikProps.handleBlur('email')}
                                    disabled={authLoading}
                                />
                            </Item>
                            <ErrorText>
                                {formikProps.touched.email &&
                                    formikProps.errors.email}
                            </ErrorText>
                            <Item regular>
                                <Input
                                    placeholder={t('placeholders.password')}
                                    onChangeText={formikProps.handleChange(
                                        'password'
                                    )}
                                    onBlur={formikProps.handleBlur('password')}
                                    secureTextEntry
                                    disabled={authLoading}
                                />
                            </Item>
                            <ErrorText>
                                {formikProps.touched.password &&
                                    formikProps.errors.password}
                            </ErrorText>
                            <ForgotPasswordText
                                onPress={() =>
                                    nav.navigate(screens.ForgotPassword)
                                }
                            >
                                {t('actions.forgotPassword')}
                            </ForgotPasswordText>
                        </CenterFormContainer>
                    </CenterRow>
                    <CenterRow>
                        <LoginButton
                            onPress={formikProps.handleSubmit}
                            disabled={authLoading}
                        >
                            {authLoading ? (
                                <UIActivityIndicator
                                    size={30}
                                    color={globalWhite}
                                />
                            ) : (
                                <Text>{t('buttons.login')}</Text>
                            )}
                        </LoginButton>
                    </CenterRow>
                </Content>
            )}
        </Formik>
    );
};

export default LoginForm;
