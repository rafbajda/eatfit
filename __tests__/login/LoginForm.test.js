import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import LoginForm from '../../app/screens/login/components/LoginForm';

describe('LoginForm snapshot', () => {
    jest.useFakeTimers();
    beforeEach(() => {
        NavigationTestUtils.resetInternalState();
    });

    it('renders component', async () => {
        const tree = renderer.create(<LoginForm />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
