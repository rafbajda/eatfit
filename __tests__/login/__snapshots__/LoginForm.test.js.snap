// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm snapshot renders component 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Array [
      Object {
        "padding": undefined,
      },
      undefined,
    ]
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutomaticScroll={true}
  enableOnAndroid={false}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  getScrollResponder={[Function]}
  handleOnScroll={[Function]}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  keyboardShouldPersistTaps="handled"
  keyboardSpace={0}
  onScroll={[Function]}
  resetKeyboardSpace={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={1}
  scrollForExtraHeightOnAndroid={[Function]}
  scrollIntoView={[Function]}
  scrollToEnd={[Function]}
  scrollToFocusedInput={[Function]}
  scrollToPosition={[Function]}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  update={[Function]}
  viewIsInsideTabBar={false}
>
  <View>
    <View
      className="sc-bxivhb dulVTN"
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        className="sc-EHOje esQQGS"
        style={
          Array [
            Object {},
            null,
          ]
        }
      >
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 1,
              "borderRadius": undefined,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
              "borderWidth": undefined,
              "flexDirection": "row",
              "marginLeft": 0,
              "opacity": 1,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            keyboardType="email-address"
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="[missing \\"en.placeholders.email\\" translation]"
            placeholderTextColor="#575757"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 50,
                "paddingLeft": 8,
                "paddingRight": 5,
                "top": 1.5,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
          />
        </View>
        <Text
          className="sc-gzVnrw iqTgrQ"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        />
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 1,
              "borderRadius": undefined,
              "borderRightWidth": 1,
              "borderTopWidth": 1,
              "borderWidth": undefined,
              "flexDirection": "row",
              "marginLeft": 0,
              "opacity": 1,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="[missing \\"en.placeholders.password\\" translation]"
            placeholderTextColor="#575757"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "fontSize": 17,
                "height": 50,
                "paddingLeft": 8,
                "paddingRight": 5,
                "top": 1.5,
              }
            }
            underlineColorAndroid="rgba(0,0,0,0)"
          />
        </View>
        <Text
          className="sc-gzVnrw iqTgrQ"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        />
        <Text
          className="sc-gZMcBi coeZUP"
          onPress={[Function]}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        >
          [missing "en.actions.forgotPassword" translation]
        </Text>
      </View>
    </View>
    <View
      className="sc-bxivhb dulVTN"
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#007aff",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 2,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 16.5,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            }
          }
          uppercase={false}
        >
          [missing "en.buttons.login" translation]
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
